atomix/cluster/src/main/java/io/atomix/cluster/messaging/impl/NettyMessagingService.java

```diff
@@ -438,11 +440,19 @@ private void initNioTransport() {
                                         connections.remove(channel);
                                       });
                             }
-                            executor.execute(() -> future.completeExceptionally(sendError));
+                            executor.execute(
+                                () -> {
+                                  future.completeExceptionally(sendError);
+                                  openFutures.remove(future);
+                                });
                           }
                         });
               } else {
-                executor.execute(() -> future.completeExceptionally(channelError));
+                executor.execute(
+                    () -> {
+                      future.completeExceptionally(channelError);
+                      openFutures.remove(future);
```

Comment:
So this was the actual bug right? That it was not removed if we have a channelError?

Reply:
Yes - also that it was removed from the list before being completed, but that had much less consequences.